src/components/FormPersister/FormPersister.tsx

```diff
@@ -0,0 +1,31 @@
+import { useContext, useEffect, useState } from 'react';
+import CircularProgress from '@mui/material/CircularProgress';
+import WizardFormContext from 'contexts/WizardFormContext';
+import { FormValuesType } from 'types/formTypes';
+
+export default function FormPersister() {
+  const { formValues, onSaveFormValues } = useContext(WizardFormContext);
+  const [isSpinnerShown, setIsSpinnerShown] = useState<boolean>(true);
+
+  useEffect(() => {
+    const localStorageData = localStorage.getItem('Context value');
+    if (localStorageData) {
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+      const parsedLocalStorageData: FormValuesType =
+        JSON.parse(localStorageData);
+      onSaveFormValues(parsedLocalStorageData);
+      setIsSpinnerShown(false);
```

Comment:
Move it out of `if`. You will need to set it to `false` anywat

Reply:
Another useful suggestions: 1. it make sense to remove value from the `localStorage` 2. Add TODO: that failure of `localStorage` manipulations should be handled; error handling not in the scope of the PR